presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.execution.QueryInfo;
+import com.facebook.presto.execution.QueryManager;
+import com.facebook.presto.execution.resourceGroups.ResourceGroupManager;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupInfo;
+import com.google.common.collect.ImmutableList;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.server.QueryStateInfo.createQueryStateInfo;
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableList;
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
+@Path("/v1/query_state")
+public class QueryStateInfoResource
+{
+    private final QueryManager queryManager;
+    private final ResourceGroupManager resourceGroupManager;
+
+    @Inject
+    public QueryStateInfoResource(
+            QueryManager queryManager,
+            ResourceGroupManager resourceGroupManager)
+    {
+        this.queryManager = requireNonNull(queryManager, "queryManager is null");
+        this.resourceGroupManager = requireNonNull(resourceGroupManager, "resourceGroupManager is null");
+    }
+
+    @GET
+    public List<QueryStateInfo> getQueryDiagnosticsByUser(@QueryParam("user") String user)
```

Comment:
I used a param because later we might change this to a list per earlier discussion with @martint. Like if an actual person has multiple "user" names associated with that person. I'm not sure whether we should change it to accept a list even now?